### PR TITLE
Move GL interface code out of the OpenGL video backend.

### PR DESCRIPTION
**Pull request by phire**
This moves the OpenGL loading/interface/context-creation code out of the OpenGL video backend into Common/GL. This means Video Software no-longer depends on the OpenGL backend being linked in to function, a dependency that made little sense.

The reason it was moved into Common/GL rather than something like VideoCommon/GL is that VideoCommon is for code common to both the OpenGL and DirectX11 backends, which GL context creation is clearly not. It is our intention in the future to potentially rename VideoCommon to something like FlipperCommon and limit it's contents to common code for the emulation the GameCube's GPU.

A longer-term goal of is to allow the creation of an OpenGL based "Big Picture" user interface that will create a OpenGL context as soon as Dolphin starts, without having to initialize the OpenGL backend first.

### CODE REVIEW

File: Source/Core/Common/GL/GLInterface/EGL.cpp [hunk 1]
```diff
@@ -2,8 +2,8 @@
 // Licensed under GPLv2+
 // Refer to the license.txt file included.
 
-#include "VideoBackends/OGL/GLInterfaceBase.h"
-#include "VideoBackends/OGL/GLInterface/EGL.h"
+#include "Common/GL/GLInterfaceBase.h"
+#include "Common/GL/GLInterface/EGL.h"
 #include "VideoCommon/DriverDetails.h"
```
Comment (degasus) [inline Source/Core/Common/GL/GLInterface/EGL.cpp]: Common must not depend on VideoCommon, else linking will fail.

File: Source/Core/Common/GL/GLInterface/EGL.h [hunk 2]
```diff
@@ -7,8 +7,8 @@
 #include <string>
 #include <EGL/egl.h>
 
+#include "Common/GL/GLInterfaceBase.h"
 #include "Core/ConfigManager.h"
```
Comment (degasus) [inline Source/Core/Common/GL/GLInterface/EGL.h]: Neither on Core

Comment (phire) [inline Source/Core/Common/GL/GLInterface/EGL.h]: I'll fix those in my next commit.

**Comment by phire** [comment]
This moves the OpenGL loading/interface/context-creation code out of the OpenGL video backend into `Common/GL`. This means Video Software no-longer depends on the OpenGL backend being linked in to function, a dependency that made little sense. 

The reason it was moved into `Common/GL` rather than something like `VideoCommon/GL` is that VideoCommon is for code common to both the OpenGL and DirectX11 backends, which GL context creation is clearly not. It is our intention in the future to potentially rename VideoCommon to something like FlipperCommon and limit it's contents to common code for the emulation the GameCube's GPU. 

A longer-term goal of is to allow the creation of an OpenGL based "BigPicture" user interface that will create a OpenGL context as soon as Dolphin starts, without having to initialize the OpenGL backend first.

File: Source/Core/Common/GL/GLInterface/EGLAndroid.cpp [hunk 3]
```diff
@@ -2,8 +2,8 @@
 // Licensed under GPLv2+
 // Refer to the license.txt file included.
 
+#include "Common/GL/GLInterface/EGLAndroid.h"
 #include "Core/Host.h"
```
Comment (degasus) [inline Source/Core/Common/GL/GLInterface/EGLAndroid.cpp]: You're also not allowed to use Core.

Comment (degasus) [inline Source/Core/Common/GL/GLInterface/EGLAndroid.cpp]: Core dependency

File: Source/Core/VideoBackends/OGL/Render.cpp [hunk 4]
```diff
@@ -24,8 +26,7 @@
 
 #include "VideoBackends/OGL/BoundingBox.h"
 #include "VideoBackends/OGL/FramebufferManager.h"
-#include "VideoBackends/OGL/GLInterfaceBase.h"
-#include "VideoBackends/OGL/GLUtil.h"
+
```
Comment (degasus) [inline Source/Core/VideoBackends/OGL/Render.cpp]: No need for an empty line.

**Comment by phire** [comment]
I've removed the final stray include and fixed up a few newline issues. 

Only two of those dependencies on VideoCommon were real, all other dependencies were just stray includes.

**Comment by waddlesplash** [comment]
LGTM

**Pull request merged**